connection_producer.go

```diff
@@ -115,13 +108,40 @@ func (c *redisDBConnectionProducer) Connection(ctx context.Context) (interface{}
 		return c.client, nil
 	}
 	var err error
+	var pem []byte
+	var poolConfig radix.PoolConfig
 
-	poolConfig := radix.PoolConfig{
-		Dialer: radix.Dialer{
-			AuthUser: c.Username,
-			AuthPass: c.Password,
-		},
+	if c.TLS {
+		pem, err = base64.StdEncoding.DecodeString(c.CaCrt)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding CaCrt")
+		}
```

Comment:
I'm thinking we can remove the requirement to base64 encode the already PEM-encoded CA cert. There are other APIs in Vault that simply take a PEM file (PEM is already base64 encoded and safe to transmit over networks). I think this provides a better UX by removing the additional step of base64 encoding the PEM file.